server/functions.lua

```diff
@@ -335,7 +334,9 @@ end
 function QBCore.Functions.HasPermission(source, permission)
     if type(permission) == "string" then
         if IsPlayerAceAllowed(source, permission) then return true end
-    elseif type(permission) == "table" then
+    end
+
+    if type(permission) == "table" then
```

Comment:
I also think this function is less readable than before. The guard clauses only improve readability when the guarded condition is the edge case and smaller than the "normal" case.

Reply:
I think for this particular case this is very debatable, this is also for extensibility, for example, if you wanted to add another requirement/check you can just do another if statement underneath and then looks far better than a series of elseif.